src/main/java/yanagishima/service/PrestoServiceImpl.java

```diff
@@ -114,226 +119,199 @@ public PrestoQueryResult doQuery(String datasource, String query, String userNam
-            csvPrinter.printRecord(columns);
-            lineNumber++;
+        try (BufferedWriter writer = Files.newBufferedWriter(resultPath, UTF_8);
+             CSVPrinter printer = new CSVPrinter(writer, CSV_FORMAT)) {
+
+            printer.printRecord(columnNames);
+            rowNumber++;
+
             while (client.isRunning()) {
-                Iterable<List<Object>> data = client.currentData().getData();
-                if (data != null) {
-                    for(List<Object> row : data) {
-                        List<String> columnDataList = new ArrayList<>();
-                        List<Object> tmpColumnDataList = row.stream().collect(Collectors.toList());
-                        for (Object tmpColumnData : tmpColumnDataList) {
-                            if (tmpColumnData instanceof Long) {
-                                columnDataList.add(((Long) tmpColumnData).toString());
-                            } else if (tmpColumnData instanceof Double) {
-                                if(Double.isNaN((Double)tmpColumnData) || Double.isInfinite((Double) tmpColumnData)) {
-                                    columnDataList.add(tmpColumnData.toString());
-                                } else {
-                                    columnDataList.add(BigDecimal.valueOf((Double) tmpColumnData).toPlainString());
-                                }
-                            } else {
-                                if (tmpColumnData == null) {
-                                    columnDataList.add(null);
-                                } else {
-                                    columnDataList.add(tmpColumnData.toString());
-                                }
-                            }
-                        }
-                        try {
-                            csvPrinter.printRecord(columnDataList);
-                            lineNumber++;
-                            resultBytes += columnDataList.toString().getBytes(StandardCharsets.UTF_8).length;
-                            if(resultBytes > maxResultFileByteSize) {
-                                String message = format("Result file size exceeded %s bytes. queryId=%s, datasource=%s", maxResultFileByteSize, queryId, datasource);
-                                storeError(db, datasource, presto.name(), client.currentStatusInfo().getId(), query, userName, message);
-                                throw new RuntimeException(message);
-                            }
-                        } catch (IOException e) {
-                            throw new RuntimeException(e);
+                Iterable<List<Object>> datum = client.currentData().getData();
+                if (datum != null) {
+                    for (List<Object> data : datum) {
+                        List<String> row = data.stream().map(TypeCoerceUtil::objectToString).collect(Collectors.toList());
```

Comment:
Is ```printer.printRecord(row);``` missing?

Reply:
Accidentally removed the most important line.